src/main/java/seedu/address/model/transaction/Timestamp.java

```diff
@@ -24,32 +36,44 @@ public class Timestamp {
     public Timestamp(String timestamp) {
         requireNonNull(timestamp);
         checkArgument(isValidTimestamp(timestamp), MESSAGE_CONSTRAINTS);
-        value = LocalDateTime.parse(timestamp);
+        value = parse(timestamp);
     }
 
+    private static LocalDateTime parse(String timestamp) {
+        if (timestamp.matches(DATE_VALIDATION + " " + TIME_VALIDATION)) {
```

Comment:
I assume this means there can only be 1 space between date and time right? If so, maybe we should add this in the datetime constraitns messagee?